exec/java-exec/src/main/java/org/apache/drill/exec/planner/physical/TopNPrel.java

```diff
@@ -115,6 +118,16 @@ public Prel addImplicitRowIDCol(List<RelNode> children) {
                                     .replace(this.getTraitSet().getTrait(DrillDistributionTraitDef.INSTANCE))
                                     .replace(collationTrait)
                                     .replace(DRILL_PHYSICAL);
-    return (Prel) this.copy(traits, children);
+    return transformTopNToSortAndLimit(children, traits, collationTrait);
+  }
+
+  private Prel transformTopNToSortAndLimit(List<RelNode> children, RelTraitSet traits, RelCollation collationTrait) {
```

Comment:
the transform is just replacing `TopNPrel` with new `SortPrel` and `PartitionLimitPrel`. My understanding is at this stage all the `SelectionVectorRemoverPrel` is also created wherever needed. In this case also between `SortPrel` and `LimitPrel` we have to insert a `SelectionVectorRemoverPrel` since `Limit` can only accept `SV2/SV None` whereas `Sort` can produce `SV None and SV4`

Reply:
SelectionVector are created at the later stage. This is called before inserting the selectionvector.